src/hotspot/share/gc/shenandoah/shenandoahFullGC.cpp

```diff
@@ -912,7 +912,6 @@ class ShenandoahPostCompactClosure : public ShenandoahHeapRegionClosure {
 
 public:
   ShenandoahPostCompactClosure() : _heap(ShenandoahHeap::heap()), _live(0) {
-    _heap->free_set()->clear();
```

Comment:
Why is that ok?

Reply:
There is no need to clear() because prepare_to_rebuild() invokes clear_internal() before it establishes new partition assignments.

Reply:
Ok.